src/status_im/contexts/wallet/send/input_amount/view.cljs

```diff
@@ -123,216 +122,222 @@
+                                                   [:wallet/wallet-send-sender-network-values])
+        receiver-network-values                   (rf/sub
+                                                   [:wallet/wallet-send-receiver-network-values])
+        token-not-supported-in-receiver-networks? (every? #(= (:type %) :not-available)
+                                                          (filter #(not= (:type %) :add)
+                                                                  receiver-network-values))
+        suggested-routes                          (rf/sub [:wallet/wallet-send-suggested-routes])
+        routes                                    (when suggested-routes
+                                                    (or (:best suggested-routes) []))
+        no-routes-found?                          (and
+                                                   (every-network-value-is-zero?
+                                                    sender-network-values)
+                                                   (not (nil? routes))
+                                                   (not loading-routes?)
+                                                   (not token-not-supported-in-receiver-networks?))
+        receiver-networks                         (rf/sub [:wallet/wallet-send-receiver-networks])
+        receiver-preferred-networks               (rf/sub
+                                                   [:wallet/wallet-send-receiver-preferred-networks])
+        receiver-preferred-networks-set           (set receiver-preferred-networks)
+        sending-to-unpreferred-networks?          (not (every? (fn [receiver-selected-network]
+                                                                 (contains?
+                                                                  receiver-preferred-networks-set
+                                                                  receiver-selected-network))
+                                                               receiver-networks))
+        limit-insufficient?                       (> (controlled-input/numeric-value input-state)
+                                                     current-limit)
+        should-try-again?                         (and (not limit-insufficient?) no-routes-found?)
+        current-address                           (rf/sub [:wallet/current-viewing-account-address])]
+    (rn/use-mount
+     (fn []
+       (let [dismiss-keyboard-fn   #(when (= % "active") (rn/dismiss-keyboard!))
+             app-keyboard-listener (.addEventListener rn/app-state "change" dismiss-keyboard-fn)]
+         #(.remove app-keyboard-listener))))
+    (rn/use-effect
+     (fn []
+       (set-input-state #(controlled-input/set-upper-limit % current-limit)))
+     [current-limit])
+    (rn/use-effect
+     (fn []
+       (when (not= input-state controlled-input/init-state)
+         (set-input-state controlled-input/init-state)))
+     [current-address])
```

Comment:
I know when `current-address` changes, the `input-state` value in the callback would also be updated. But I think for correctness, we can use the value for `input-state` from the callback for setState in `set-input-state`.

That would be something like this:

```cljs
(rn/use-effect
         (fn []
           (set-input-state #(if (= % controlled-input/init-state)
                               %
                               controlled-input/init-state)))
         [current-address])
```

What do you think?

Reply:
yeah makes sense, (just as a nit I'll use a `fn` so that I name the value `prev-state` so it's very clear. but overall sounds good to me! :)

thanks!

Reply:
actually looking at this, I'm not sure it really makes sense. 
Both will set the state to init/state. I don't want to trigger any state change if the initial value is already set that way. What you are suggesting here is a bit different, no?

Reply:
When you return the same value that was in the prev state, it doesn't trigger an update, because the reference stays the same. So the only time a state update is triggered is in the second case where the previous value is not equal to init-state.

Reply:
true, I forgot about that. Anyway, I updated the function to take a different approach and it clears it safely now 👍